File: idealista-button-extension/generate_icon.py
#!/usr/bin/env python3
"""
Script para gerar ícones da extensão ImobFlash Agent
Gera ícones nos tamanhos: 16x16, 48x48, 128x128
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size):
    """Cria um ícone no tamanho especificado"""
    # Cria imagem com fundo transparente
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Cores do gradiente (roxo/azul)
    color1 = (102, 126, 234)  # #667eea
    color2 = (118, 75, 162)   # #764ba2
    
    # Desenha fundo com gradiente circular
    center = size // 2
    radius = size // 2 - 2
    
    # Cria gradiente simples (círculo preenchido)
    for i in range(radius):
        # Interpolação de cor
        ratio = i / radius
        r = int(color1[0] * (1 - ratio) + color2[0] * ratio)
        g = int(color1[1] * (1 - ratio) + color2[1] * ratio)
        b = int(color1[0] * (1 - ratio) + color2[0] * ratio)
        
        # Desenha círculo concêntrico
        draw.ellipse(
            [center - radius + i, center - radius + i, 
             center + radius - i, center + radius - i],
            fill=(r, g, b, 255)
        )
    
    # Desenha borda branca sutil
    draw.ellipse(
        [2, 2, size - 2, size - 2],
        outline=(255, 255, 255, 200),
        width=max(1, size // 32)
    )
    
    # Desenha ícone de robô/casa
    # Cabeça do robô (retângulo arredondado)
    head_size = int(size * 0.4)
    head_x = center - head_size // 2
    head_y = center - head_size // 2 - int(size * 0.1)
    
    # Corpo (casa)
    body_width = int(size * 0.5)
    body_height = int(size * 0.35)
    body_x = center - body_width // 2
    body_y = center + int(size * 0.05)
    
    # Desenha casa (corpo do robô)
    # Telhado (triângulo)
    roof_points = [
        (center, body_y - int(size * 0.15)),
        (body_x, body_y),
        (body_x + body_width, body_y)
    ]
    draw.polygon(roof_points, fill=(255, 255, 255, 255))
    
    # Corpo da casa (retângulo)
    draw.rectangle(
        [body_x, body_y, body_x + body_width, body_y + body_height],
        fill=(255, 255, 255, 255)
    )
    
    # Porta (retângulo pequeno)
    door_width = int(size * 0.15)
    door_height = int(size * 0.2)
    door_x = center - door_width // 2
    door_y = body_y + body_height - door_height
    draw.rectangle(
        [door_x, door_y, door_x + door_width, door_y + door_height],
        fill=(102, 126, 234, 255)
    )
    
    # Olhos do robô (dois círculos pequenos no telhado)
    eye_size = max(2, size // 16)
    eye_y = body_y - int(size * 0.1)
    draw.ellipse(
        [center - int(size * 0.12) - eye_size, eye_y - eye_size,
         center - int(size * 0.12) + eye_size, eye_y + eye_size],
        fill=(102, 126, 234, 255)
    )
    draw.ellipse(
        [center + int(size * 0.12) - eye_size, eye_y - eye_size,
         center + int(size * 0.12) + eye_size, eye_y + eye_size],
        fill=(102, 126, 234, 255)
    )
    
    return img

def main():
    """Gera todos os ícones necessários"""
    sizes = [16, 48, 128]
    icons_dir = 'icons'
    
    # Cria diretório de ícones se não existir
    os.makedirs(icons_dir, exist_ok=True)
    
    print("🎨 Gerando ícones da extensão ImobFlash Agent...")
    
    for size in sizes:
        icon = create_icon(size)
        filename = f'{icons_dir}/icon{size}.png'
        icon.save(filename, 'PNG')
        print(f"✅ Criado: {filename} ({size}x{size})")
    
    print("\n✨ Ícones gerados com sucesso!")
    print(f"📁 Diretório: {icons_dir}/")

if __name__ == '__main__':
    main()

File: idealista-button-extension/generate_icon.js
#!/usr/bin/env node
/**
 * Script para gerar ícones da extensão ImobFlash Agent
 * Requer: npm install canvas (ou usar ferramenta online)
 * 
 * Alternativa: Use https://convertio.co/svg-png/ para converter icon.svg
 * para os tamanhos: 16x16, 48x48, 128x128
 */

const fs = require('fs');
const path = require('path');

// Verifica se canvas está disponível
let Canvas;
try {
    Canvas = require('canvas');
} catch (e) {
    console.log('⚠️  Biblioteca "canvas" não encontrada.');
    console.log('📝 Instale com: npm install canvas');
    console.log('📝 Ou use o arquivo icon.svg e converta online em: https://convertio.co/svg-png/');
    console.log('\n✨ Tamanhos necessários: 16x16, 48x48, 128x128');
    process.exit(1);
}

function createIcon(size) {
    const canvas = Canvas.createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // Fundo transparente
    ctx.clearRect(0, 0, size, size);
    
    // Gradiente circular
    const center = size / 2;
    const radius = size / 2 - 2;
    const gradient = ctx.createRadialGradient(center, center, 0, center, center, radius);
    gradient.addColorStop(0, '#667eea');
    gradient.addColorStop(1, '#764ba2');
    
    // Círculo de fundo
    ctx.beginPath();
    ctx.arc(center, center, radius, 0, Math.PI * 2);
    ctx.fillStyle = gradient;
    ctx.fill();
    ctx.strokeStyle = 'rgba(255, 255, 255, 0.3)';
    ctx.lineWidth = Math.max(1, size / 64);
    ctx.stroke();
    
    // Casa (corpo do robô)
    const houseWidth = size * 0.5;
    const houseHeight = size * 0.35;
    const houseX = center - houseWidth / 2;
    const houseY = center + size * 0.05;
    
    // Telhado (triângulo)
    ctx.beginPath();
    ctx.moveTo(center, houseY - size * 0.15);
    ctx.lineTo(houseX, houseY);
    ctx.lineTo(houseX + houseWidth, houseY);
    ctx.closePath();
    ctx.fillStyle = 'white';
    ctx.fill();
    
    // Corpo da casa
    ctx.fillStyle = 'white';
    ctx.fillRect(houseX, houseY, houseWidth, houseHeight);
    
    // Porta
    const doorWidth = size * 0.15;
    const doorHeight = size * 0.2;
    const doorX = center - doorWidth / 2;
    const doorY = houseY + houseHeight - doorHeight;
    ctx.fillStyle = '#667eea';
    ctx.fillRect(doorX, doorY, doorWidth, doorHeight);
    
    // Olhos do robô (no telhado)
    const eyeSize = Math.max(2, size / 16);
    const eyeY = houseY - size * 0.1;
    ctx.fillStyle = '#667eea';
    
    // Olho esquerdo
    ctx.beginPath();
    ctx.arc(center - size * 0.12, eyeY, eyeSize, 0, Math.PI * 2);
    ctx.fill();
    
    // Olho direito
    ctx.beginPath();
    ctx.arc(center + size * 0.12, eyeY, eyeSize, 0, Math.PI * 2);
    ctx.fill();
    
    return canvas;
}

function main() {
    const sizes = [16, 48, 128];
    const iconsDir = path.join(__dirname, 'icons');
    
    // Cria diretório se não existir
    if (!fs.existsSync(iconsDir)) {
        fs.mkdirSync(iconsDir, { recursive: true });
    }
    
    console.log('🎨 Gerando ícones da extensão ImobFlash Agent...\n');
    
    sizes.forEach(size => {
        const icon = createIcon(size);
        const filename = path.join(iconsDir, `icon${size}.png`);
        const buffer = icon.toBuffer('image/png');
        fs.writeFileSync(filename, buffer);
        console.log(`✅ Criado: icons/icon${size}.png (${size}x${size})`);
    });
    
    console.log('\n✨ Ícones gerados com sucesso!');
    console.log(`📁 Diretório: ${iconsDir}/`);
}

main();

